packages/js/navigation/src/index.js

```diff
@@ -266,6 +267,42 @@ export const addHistoryListener = ( listener ) => {
 	};
 };
 
+/**
+ * Determines if a URL is a WC admin url.
+ *
+ * @param {*} url - the url to test
+ * @return {boolean} true if the url is a wc-admin URL
+ */
+export const isWCAdmin = ( url = window.location.href ) => {
+	return /admin.php\?page=wc-admin/.test( url );
+};
+
+/**
+ * A utility function that navigates to a page, using a redirect
+ * or the router as appropriate.
+ *
+ * @param {Object} args        - All arguments.
+ * @param {string} [args.url]  - Relative or absolute url to navigate to.
+ * @param {string} [args.path] - Path to navigate to.
+ */
+export const navigateTo = ( { url, path } ) => {
+	if ( url && url.startsWith( 'http' ) ) {
+		window.location.href = url;
+		return;
+	}
+
+	const newPath = path ? path : getNewPath( {}, url, {} );
+
+	if ( ! isWCAdmin() ) {
```

Comment:
Do we need to pass `newPath` here?  Otherwise it seems like it will check the current page instead of the URL in question.

Reply:
Good thought @joshuatf . For my current use case it was actually necessary to test the current URL, since I'm handling the case of navigating _from_ an embedded page to a WCAdmin page. That said, I should be checking both the origin and destination, which I've added.